### DISCUSSION TITLE
Library initialization

### DISCUSSION THREAD

**Discussion by steamstreetjon** [aws/aws-sdk-kotlin#603] [General]
Typically in Lambda, we initialize service objects outside of the main handler. In the Java SDK, this means just creating a client like:

```
val sqs = SqsClient.create()
```

But for Kotlin, we are using the `fromEnvironment` call, which is suspended, so we can't just call it, and have to wrap it in some kind of coroutine context, which means something like:

```
private val eventBridge = runBlocking { EventBridgeClient.fromEnvironment() }
```

This is fine, but pretty cumbersome. Is there a preferred way to do something like this that doesn't require the runBlocking?

**Comment by ianbotsf** [comment]
Hi @steamstreetjon, thanks for the question.

The example code you provided _is_ the preferred way to synchronously call one of our `suspend` methods. Because initializing a client may involve network calls depending on your environment (e.g., retrieving credentials from STS) we prefer to give callers control over how to sequence that. If you don't desire asynchrony in this case and don't mind blocking the thread until the call completes, wrapping in `runBlocking` is the solution.

I'd like to understand more about what you find cumbersome with the `runBlocking` wrapper. Is it merely that there is more code? Is there another mechanism you'd prefer for synchronously invoking calls that (may) reach out to the network?
